content/blog/recipes-1-3-0/index.Rmd

```diff
@@ -0,0 +1,183 @@
+
+```{r}
+library(modeldata)
+tate_text
+```
+
+We are loading the modeldata package to get `tate_text` which has a character column `title`. If we don't do anything then it turns into a factor.
+
+```{r}
+recipe(~., data = tate_text) |>
+  prep() |>
+  bake(tate_text)
+```
+
+But we can set `strings_as_factors = FALSE` in `recipe()` and it won't anymore.
+
+```{r}
+recipe(~., data = tate_text, strings_as_factors = FALSE) |>
+  prep() |>
+  bake(tate_text)
+```
+
+This change should also make pragmatic sense as whether you want to turn strings into factors is something that should encoded into the recipe itself.
+
+## Deprecating `step_select()`
+
+We have started the process of deprecating `step_select()`. Given the number of issues people are having with the step and the fact that it doesn't play well with workflows we think this is the right call.
+
+There are two main use cases where `step_select()` was used. Removing variables, and selecting variables. Removing variables when done with `-` in `step_select()`
+
+```{r, warning=FALSE}
+recipe(mpg ~ ., mtcars) |>
+  step_select(-starts_with("d")) |>
+  prep() |>
+  bake(new_data = NULL)
+```
+
+These use cases can seamlessly be converted to use `step_rm()` without the `-` for the same result.
+
+```{r}
+recipe(mpg ~ ., mtcars) |>
+  step_rm(starts_with("d")) |>
+  prep() |>
+  bake(new_data = NULL)
+```
+
+For selecting variables there are two cases. The first is as a tool to select which variables to use in our model. We recommend that you use `select()` to do that before passing the data into the `recipe()`. This is especially helpful since [recipes are tighter with respect to their input types](https://www.tidyverse.org/blog/2024/07/recipes-1-1-0/#column-type-checking), so only passing the data you need to use is helpful.
+
+If you need to do the selection after another step takes effect you should still be able to do so, by using `step_rm()` in the following manner.
+
+```r
+step_rm(recipe, all_predictors(), -all_of(<variables that you want to keep>))
+```
+
+## `step_dummy()` contrasts argument
+
+Contrasts such as `contr.treatment()` and `contr.poly()` are used in `step_dummy()` to determine how the steps should translate categorical values into one or more numeric columns. Traditionally the contrasts were set using `options()` like so:
+
+```{r}
+options(contrasts = c(unordered = "contr.poly", ordered = "contr.poly"))
+```
+
+```{r, warning=FALSE}
+recipe(~species + island, penguins) |>
+  step_dummy(all_nominal_predictors()) |>
+  prep() |>
+  bake(new_data = penguins)
+```
+
+The issue with this approach is that it pulls from `options()` when it needs it instead of storing the information. This means that if you put this recipe in production you will need to set the option in the production environment to match that of the training environment.
+
+```{r}
+#| echo: false
+options(contrasts = c(unordered = "contr.treatment", ordered = "contr.poly"))
```

Comment:
Can you add a note on how to do this with a contrast in an external package?

Reply:
done in https://github.com/tidyverse/tidyverse.org/pull/730/commits/18ee2a6c57084c1317996050fa3fe46cf716a931